src/hub.js

```diff
@@ -1,22 +1,54 @@
 import uuid from 'uuid';
 import { log } from './log';
 import { app } from './app';
-import { postMessage, hubMessageValid, publishMessageValid } from './msg';
-import { HEARTBEAT } from './lib';
+import {
+	postMessage,
+	hubMessageValid,
+	publishMessageValid,
+	spawnMessageValid
+} from './msg';
+import { HEARTBEAT, CHROME_APP_ID } from './lib';
+
+let hubInstance;
 
 /**
  * Special features supplied by the Tradeshift® Chrome™.
  * @typedef {object} ChromeWindowFeatures
  * @property {function(Window): string} appIdByWindow Called to get an appId based on a Window object.
  * @property {function(string, Window): Window} windowByAppId Called to get a window object based on an appId and the requesting app's Window object.
- * @property {function(Window): void} appTimeout Called when an app fails to reply in time to a PING request.
+ * @property {function(Window, string): void} appTimeout Called when an app fails to reply in time to a PING request.
+ * @property {object: void} handlers Contains all handlers for SPAWN, etc.
  */
 
 /**
  * The Message Broker AKA The Hub.
  * @param {ChromeWindowFeatures} chrome Special features supplied by the Tradeshift® Chrome™
  */
 export function hub(chrome) {
+	if (hubInstance) {
+		return hubInstance;
+	}
+
+	const { appIdByWindow, windowByAppId, appTimeout, handlers } = chrome;
+
+	/**
+	 * Quickly test that appIdByWindow & windowByAppId work for 'Tradeshift.Chrome'
+	 */
+	{
```

Comment:
Is this just for scoping?

Reply:
yes, we don't need these variables later, the scope should allow for garbage collection.